src/main/scala/rifs/business/controllers/ApplicationController.scala

```diff
@@ -162,26 +162,37 @@ class ApplicationController @Inject()(val cached: Cached, applications: Applicat
   val RIFS_EMAIL = "rifs.email"
   val RIFS_DUMMY_APPLICANT_EMAIL = s"$RIFS_EMAIL.dummyapplicant"
   val RIFS_REPLY_TO_EMAIL = s"$RIFS_EMAIL.replyto"
+  val RIFS_DUMMY_MANAGER_EMAIL = s"$RIFS_EMAIL.dummymanager"
 
   def submit(id: ApplicationId) = Action.async { _ =>
-    applications.submit(id).flatMap {
-      case Some(submissionRef) =>
-        val res = JsObject(Seq("applicationRef" -> Json.toJson(submissionRef)))
-        val from = config.underlying.getString(RIFS_REPLY_TO_EMAIL)
-        val to = config.underlying.getString(RIFS_DUMMY_APPLICANT_EMAIL)
-
-        notifications.notifyPortfolioManager(submissionRef, from, to).map {
-          _.map { _ => res } // this would wait for e-mail to be sent, we can just put it onto threadpool
-        }.map {
-          jsonResult(_)
-        }.recover {
-          case t =>
-            Logger.error("Failed to send email on application submission", t)
-            Ok(res) // we return OK, as the application was submitted
-        }
-      case None => Future {
-        jsonResult[ApplicationId](None)
-      }
+
+      applications.submit(id).flatMap {
+        case Some(submissionRef) =>
+          val res = JsObject( Seq("applicationRef" -> Json.toJson(submissionRef) ) )
+          val from = config.underlying.getString(RIFS_REPLY_TO_EMAIL)
+          val to = config.underlying.getString(RIFS_DUMMY_APPLICANT_EMAIL)
+          val mgrEmail = config.underlying.getString(RIFS_DUMMY_MANAGER_EMAIL)
+          // now send notifications
+          Seq(
+              ("Manager", notifications.notifyPortfolioManager(submissionRef, from, to) ),
+              ("Applicant", notifications.notifyApplicant(submissionRef, DateTime.now(DateTimeZone.UTC), from, to, mgrEmail) )
+          )
+          .foldLeft(Future.successful(res)) {
```

Comment:
I think this is really over-complicated here for no good reason. It's very hard to see how this is going to behave if, say, the first call in the Seq() fails. Does the second one proceed? Rather than using a Seq and a fold, why not just make the two calls and combine the results?